From 8fd9f52c178cb55ea8faf3a56b671fd562c76b5d Mon Sep 17 00:00:00 2001
From: Marc Mutz <marc.mutz@qt.io>
Date: Thu, 4 Sep 2025 14:42:11 +0200
Subject: [PATCH 046/553] [docs] mention why one would want
 QTEST_THROW_ON_{FAIL,SKIP}

(The compile-time switch, as opposed to the runtime switch via
environment variables.)

It's mainly so you can use the QCOMPARE/QVERIFY/QSKIP macros in
functions that don't return void.

Make sure the documentation reflects that.

Amends e769cf026e328ed7fff660c204ce6e55b80114e3.

Pick-to: 6.9 6.8
Change-Id: I44b15899a8ab57d2cedc0318cc7c2a6859f643e7
Reviewed-by: Edward Welbourne <edward.welbourne@qt.io>
(cherry picked from commit e8a0014cf292ddb170c598c5f1f7de4fb47132b1)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/testlib/qtestcase.cpp | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/src/testlib/qtestcase.cpp b/src/testlib/qtestcase.cpp
index 7f07ac5213a..000daa18188 100644
--- a/src/testlib/qtestcase.cpp
+++ b/src/testlib/qtestcase.cpp
@@ -204,6 +204,18 @@ void Internal::maybeThrowOnSkip()
         Internal::throwOnSkip();
 }
 
+/*
+//! [return-void]
+    Defining this macro is useful if you wish to use \1 in functions that have
+    a non-\c{void} return type. Without this macro defined, \2, for example,
+    expands to a statement including \c{return;}, so it cannot be used in
+    functions (or lambdas) that return something else than \c{void}, e.g.
+    \l{QString}. This includes the case where throwing exceptions is enabled
+    only at runtime (using \3). With this macro defined, \2 instead expands to
+    a statement without a \c{return;}, which is usable from any function.
+//! [return-void]
+*/
+
 /*!
     \since 6.8
     \macro QTEST_THROW_ON_FAIL
@@ -211,6 +223,8 @@ void Internal::maybeThrowOnSkip()
 
     When defined, QCOMPARE()/QVERIFY() etc always throw on failure.
     QTest::setThrowOnFail() then no longer has any effect.
+
+    \include qtestcase.cpp {return-void} {QCOMPARE() or QVERIFY()} {QCOMPARE()} {QTest::setThrowOnFail(true)}
 */
 
 /*!
@@ -220,6 +234,8 @@ void Internal::maybeThrowOnSkip()
 
     When defined, QSKIP() always throws. QTest::setThrowOnSkip() then no longer
     has any effect.
+
+    \include qtestcase.cpp {return-void} {QSKIP()} {QSKIP()} {QTest::setThrowOnSkip(true)}
 */
 
 /*!
-- 
2.51.2

